internal/services/events_test.go

```diff
@@ -24,3 +42,344 @@ func TestProcessEvent(t *testing.T) {
+		StakingAmount:             10_000,
+		StakingOutputIdx:          0,
+		StakingBTCTimestamp:       1753970681,
+		StakerBtcPkHex:            "3f8f4496a7367a7c3fe78f95c084578b228e20325697cfe423936b905f7ac062",
+		StakerBabylonAddress:      "bbn1dppj9xellvzrh7x60vft4u8cpkyrvv3camt8ps",
+		FinalityProviderBtcPksHex: []string{"c384e26491dfec5e021a292a5f3b9b21e3c7aed611d0ecd3a96fd63b8e7e09ab"},
+		StartHeight:               263034,
+		EndHeight:                 323034,
+		State:                     types.StateExpansion,
+		SubState:                  types.SubStateEarlyUnbonding,
+		StateHistory: []model.StateRecord{
+			{
+				State:        types.StatePending,
+				BbnHeight:    1347,
+				BbnEventType: "EventBTCDelegationCreated",
+			},
+			{
+				State:        types.StateVerified,
+				BbnHeight:    1348,
+				BbnEventType: "EventCovenantQuorumReached",
+			},
+			{
+				State:        types.StateActive,
+				BbnHeight:    1513,
+				BbnEventType: "EventBTCDelegationInclusionProofReceived",
+			},
+			{
+				State:        types.StateExpansion,
+				SubState:     types.SubStateEarlyUnbonding,
+				BbnHeight:    2165,
+				BbnEventType: "EventBTCDelgationUnbondedEarly",
+			},
+		},
+		ParamsVersion:         0,
+		UnbondingTime:         20,
+		UnbondingTx:           "0200000001630cbc4f6d89ccd754bb133c2ac22e09594ff65d4e58fdf3277859332426439f0000000000ffffffff01581b000000000000225120371fbc82a29d9b0be545f11444768dc8534f2d24b3c7443f54150a446217a0a400000000",
+		UnbondingStartHeight:  263048,
+		UnbondingBTCTimestamp: 1753977985,
+		CovenantSignatures: []model.CovenantSignature{
+			{
+				CovenantBtcPkHex: "a5c60c2188e833d39d0fa798ab3f69aa12ed3dd2f3bad659effa252782de3c31",
+				SignatureHex:     "42d3d487401721b05e8562c83290954d8b5ef03c232895c95eb2695ecc41b946d6048fbc276610d0aa5c300dd2a81d3a4ee5aced0d69bd9f6852abb0875b064f",
+			},
+			{
+				CovenantBtcPkHex: "ffeaec52a9b407b355ef6967a7ffc15fd6c3fe07de2844d61550475e7a5233e5",
+				SignatureHex:     "1d643f3ff8d3bf4146f0bf98a6d5c724706f774345b7790f240e38000c9ee63e8af8d7bd54512b1e747ee3d1758df294dd05e9fbbb8cde5da927fbb6a27dbe89",
+			},
+			{
+				CovenantBtcPkHex: "59d3532148a597a2d05c0395bf5f7176044b1cd312f37701a9b4d0aad70bc5a4",
+				SignatureHex:     "4e8e9e381ed8a8440cb7f70b93fcfbef2b563c31bc04faec9777b39a22fd417ae58a5a5d6d5f40a701fabed8c8543d75ab3656f9a4308017be7c42f44fce82f6",
+			},
+		},
+		BTCDelegationCreatedBlock: model.BTCDelegationCreatedBbnBlock{
+			Height:    1347,
+			Timestamp: 1753970253,
+		},
+		SlashingTx:               model.SlashingTx{},
+		PreviousStakingTxHashHex: "",
+	}
+	assert.Equal(t, expectedDelegation, delegation)
+
+	t.Run("expiry check", func(t *testing.T) {
+		// this subtest checks that after checkExpiry(ctx) call (it's called by poller)
+		// state of the delegation is unmodified
+		tipHeight := uint64(263657)
+		btc.On("GetTipHeight", ctx).Return(tipHeight, nil)
+
+		//eventConsumer.On("PushWithdrawableStakingEvent", internalCtx, &client.StakingEvent{
```

Comment:
I keep this commented code so you can check the current flow, what would happen if new changes weren't here